lib/api/CSingleStreamSearcher.cc

```diff
@@ -14,6 +14,17 @@ CSingleStreamSearcher::CSingleStreamSearcher(const TIStreamP& stream)
     : m_Stream(stream) {
 }
 
+CSingleStreamSearcher::~CSingleStreamSearcher() {
+    this->consumeStream();
```

Comment:
It would be good to add a comment here to say this is because the Elasticsearch side might send more docs than are required, and will block if we don't read them.